test/uk/gov/hmrc/digitalengagementplatformskin/javascripts/utils/ChatContainerEventListener.spec.js

```diff
@@ -0,0 +1,85 @@
+import ChatContainer from '../../../../../../../app/assets/javascripts/utils/ChatContainer';
+
+import { ContainerHtml as EmbeddedContainerHtml } from '../../../../../../../app/assets/javascripts/views/embedded/EmbeddedContainerHtml';
+import { ContainerHtml as PopupContainerHtml } from '../../../../../../../app/assets/javascripts/views/popup/PopupContainerHtml';
+
```

Comment:
you could potentially use [beforeEach()](https://jestjs.io/docs/setup-teardown) to do this:
`let chatContainer;

  beforeEach(() => {
    chatContainer = new ChatContainer(null, EmbeddedContainerHtml, null);
  });` 

then in each case, you would do `jest.spyOn(chatContainer.eventHandler, 'onSend');`

Reply:
I think the issue is we have to use a different bit of HTML dependent on which test it is

Reply:
and accounting for that is probably writing more code than it is cleaning up. fair enough! 😄